### PR TITLE
switch to commit to avoid write limitations

### LINKED ISSUE #126: Use Firestore.commit call instead of Firestore.write
`write` call is limited to 10 simultaneous writes. For writing large number of entities this introduces a bottleneck. 

`commit` call is the recommended way to do that.

### PR DESCRIPTION
**Pull request by dmitry-s**
fixes #126

### CODE REVIEW

**Review by ttomsu — APPROVED** [review]

File: spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/spring/data/firestore/FirestoreTemplate.java [hunk 1]
```diff
@@ -209,9 +211,17 @@ public boolean isUsingStreamTokens() {
 				return Flux.from(instances).doOnNext(t -> writes.add(createUpdateWrite(t)));
 			}
 
-			Flux<List<T>> inputs = Flux.from(instances).bufferTimeout(this.writeBufferSize, this.writeBufferTimeout);
-			return ObservableReactiveUtil.streamingBidirectionalCall(
-					this::openWriteStream, inputs, this::buildWriteRequest);
+			return Flux.from(instances).bufferTimeout(this.writeBufferSize, this.writeBufferTimeout)
+					.flatMap(ts -> {
```
Comment (elefeint) [inline spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/spring/data/firestore/FirestoreTemplate.java:215]: I keep reading it as "timestamp". Rename to something more meaningful?

File: spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/spring/data/firestore/FirestoreTemplate.java [hunk 2]
```diff
@@ -291,10 +301,17 @@ public FirestoreMappingContext getMappingContext() {
 				//In a transaction, all write operations should be sent in the commit request, so we just collect them
 				return Flux.from(documentNames).doOnNext(t -> writes.add(createDeleteWrite(t)));
 			}
-			return ObservableReactiveUtil.streamingBidirectionalCall(
-					this::openWriteStream,
-					documentNames.bufferTimeout(this.writeBufferSize, this.writeBufferTimeout),
-					this::buildDeleteRequest);
+			return Flux.from(documentNames).bufferTimeout(this.writeBufferSize, this.writeBufferTimeout)
+					.flatMap(ts -> {
```
Comment (elefeint) [inline spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/spring/data/firestore/FirestoreTemplate.java:305]: Same, rename `ts` to something -- it also turns into `e` for Entity later in this block, so maybe something more consistent like "entity".

File: spring-cloud-gcp-data-firestore/src/test/java/com/google/cloud/spring/data/firestore/it/FirestoreIntegrationTests.java [hunk 3]
```diff
@@ -221,8 +222,8 @@ public void saveAllTest() {
 	@Test
 	public void saveAllBulkTest() {
 		Flux<User> users = Flux.create(sink -> {
-			for (int i = 0; i < 1000; i++) {
-				sink.next(new User("testUser " + i, i));
+			for (int i = 0; i < 10000; i++) {
```
Comment (elefeint) [inline spring-cloud-gcp-data-firestore/src/test/java/com/google/cloud/spring/data/firestore/it/FirestoreIntegrationTests.java:225]: We can probably keep it at 1,000. 10K was useful for weeding out flakes, but likely not necessary for a regularly run test.

**Review by elefeint — CHANGES_REQUESTED** [review]
Small naming things, otherwise lgtm.

**Review by meltsufin — APPROVED** [review]

**Review by elefeint — APPROVED** [review]

**Pull request merged**